source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -75,6 +75,25 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(
       random_generator_(context.api().randomGenerator()) {
   // TODO(boteng): introduce additional validation for the authority and path headers
   // based on the default UHV when it is available.
+
+  // Process send payload.
+  if (config.http_health_check().has_send()) {
+    Protobuf::RepeatedPtrField<envoy::config::core::v3::HealthCheck::Payload> send_repeated;
+    send_repeated.Add()->CopyFrom(config.http_health_check().send());
+    auto send_bytes_or_error = PayloadMatcher::loadProtoBytes(send_repeated);
```

Comment:
There is some historical reason that this `loadProtoBytes` will only receive repeated payload, while the send of health checker is non-repeated proto. It would be nice to do some refactor here, not a blocker though.

Reply:
Sure! I can do that as a follow-up.